<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="Arm_URDF_2025">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0.06 0.06 0.00437247635063724"
        rpy="0 0 0" />
      <mass
        value="0.124661946710585" />
      <inertia
        ixx="0.000147804282431458"
        ixy="-2.46473848275475E-20"
        ixz="-5.95512543966335E-22"
        iyy="0.000147804282431458"
        iyz="1.60346272184515E-21"
        izz="0.000293922860163162" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="link_1">
    <inertial>
      <origin
        xyz="-5.48442406074559E-07 -7.17314944642876E-17 0.0336041265825655"
        rpy="0 0 0" />
      <mass
        value="0.137634968166129" />
      <inertia
        ixx="0.000446367257769463"
        ixy="-2.36293542467549E-19"
        ixz="3.10211596816188E-09"
        iyy="0.000294001564310723"
        iyz="3.2777969911918E-19"
        izz="0.000373994827351754" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_1"
    type="continuous">
    <origin
      xyz="0.06 0.06 0.0135"
      rpy="0 0 1.57" />
    <parent
      link="base_link" />
    <child
      link="link_1" />
    <axis
      xyz="0 0 -1" />
  </joint>
  <link
    name="link_2">
    <inertial>
      <origin
        xyz="-0.0966215327084988 0.0316225 0.148851302099507"
        rpy="0 0 0" />
      <mass
        value="1.79663833155198" />
      <inertia
        ixx="0.0329129866831077"
        ixy="4.55700567273967E-19"
        ixz="0.00437798607290387"
        iyy="0.0349483664750651"
        iyz="-1.58372106889851E-18"
        izz="0.0032331177263641" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_2"
    type="continuous">
    <origin
      xyz="1.24999999999858E-05 -0.03162775 0.0747000000000002"
      rpy="0 0 0" />
    <parent
      link="link_1" />
    <child
      link="link_2" />
    <axis
      xyz="0 -1 0" />
  </joint>
  <link
    name="link_3">
    <inertial>
      <origin
        xyz="0.022527417749687 0.036975 0.0885307406770253"
        rpy="0 0 0" />
      <mass
        value="0.622999243464079" />
      <inertia
        ixx="0.00200228691765054"
        ixy="6.55630474936292E-20"
        ixz="-0.000294247168897306"
        iyy="0.0020516448574634"
        iyz="-5.6058962694986E-19"
        izz="0.00138625401056328" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_3"
    type="continuous">
    <origin
      xyz="-0.112992 0 0.384077"
      rpy="0 0 0" />
    <parent
      link="link_2" />
    <child
      link="link_3" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="link_4">
    <inertial>
      <origin
        xyz="0.10524344492505 -7.05797014825815E-08 0"
        rpy="0 0 0" />
      <mass
        value="0.970122911594951" />
      <inertia
        ixx="0.00243457030866732"
        ixy="4.63516303360271E-09"
        ixz="-1.67537061699453E-19"
        iyy="0.00385833789304732"
        iyz="-8.80629759801178E-19"
        izz="0.00256309318654037" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_4"
    type="continuous">
    <origin
      xyz="0.084452 0.036975 0.115"
      rpy="0 0 0" />
    <parent
      link="link_3" />
    <child
      link="link_4" />
    <axis
      xyz="1 0 0" />
  </joint>
  <link
    name="link_5">
    <inertial>
      <origin
        xyz="-0.0198884732871961 0.0406401113330525 0.00629862989332275"
        rpy="0 0 0" />
      <mass
        value="0.284717288401274" />
      <inertia
        ixx="0.000299965546841619"
        ixy="-1.58843113349382E-09"
        ixz="-8.63807918048834E-05"
        iyy="0.000511230229605918"
        iyz="-2.43986556603002E-09"
        izz="0.000420975876271412" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_5.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_5.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_5"
    type="continuous">
    <origin
      xyz="0.305479 -0.0406420000000001 0"
      rpy="0 0 0" />
    <parent
      link="link_4" />
    <child
      link="link_5" />
    <axis
      xyz="0 -1 0" />
  </joint>
  <link
    name="link_6">
    <inertial>
      <origin
        xyz="0.00692827119863587 3.3907380514453E-09 1.11022302462516E-16"
        rpy="0 0 0" />
      <mass
        value="0.175170524925624" />
      <inertia
        ixx="0.000293643764169239"
        ixy="-1.03961089729387E-11"
        ixz="2.2784964648173E-20"
        iyy="5.11611215542502E-05"
        iyz="-1.11398912388781E-20"
        izz="0.00026066688002045" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_6.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/link_6.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_6"
    type="continuous">
    <origin
      xyz="0.032 0.04064 0"
      rpy="0 0 0" />
    <parent
      link="link_5" />
    <child
      link="link_6" />
    <axis
      xyz="1 0 0" />
  </joint>
  <link
    name="finger_1">
    <inertial>
      <origin
        xyz="0.0855523666641121 0.00980197809754374 -0.015"
        rpy="0 0 0" />
      <mass
        value="0.131585801201098" />
      <inertia
        ixx="2.06584333808883E-05"
        ixy="-1.27622966682436E-06"
        ixz="2.59268411809251E-20"
        iyy="0.000250311821820243"
        iyz="-7.77707182589073E-22"
        izz="0.000250190068564209" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/finger_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/finger_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="finger_joint_1"
    type="continuous">
    <origin
      xyz="0.0342469999999999 0.0690450000000001 0.0150000000000005"
      rpy="0 0 0" />
    <parent
      link="link_6" />
    <child
      link="finger_1" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="finger_2">
    <inertial>
      <origin
        xyz="0.0855523666641121 -0.00980197809754374 -0.0150000000000001"
        rpy="0 0 0" />
      <mass
        value="0.131585801201098" />
      <inertia
        ixx="2.06584333808883E-05"
        ixy="1.2762296668244E-06"
        ixz="4.6284791662046E-20"
        iyy="0.000250311821820244"
        iyz="4.87544459169818E-21"
        izz="0.00025019006856421" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/finger_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/finger_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="finger_joint_2"
    type="continuous">
    <origin
      xyz="0.0342469999999999 -0.0690460000000001 0.015"
      rpy="0 0 0" />
    <parent
      link="link_6" />
    <child
      link="finger_2" />
    <axis
      xyz="0 0 -1" />
  </joint>
  <link
    name="camera_link">
    <inertial>
      <origin
        xyz="0.0237078925078273 0.00399999999999998 0.00849517559098378"
        rpy="0 0 0" />
      <mass
        value="0.0455289011770119" />
      <inertia
        ixx="1.08276639266657E-05"
        ixy="1.63536654407214E-21"
        ixz="-2.23258863162784E-06"
        iyy="8.28802476045037E-06"
        iyz="-3.16596447731408E-21"
        izz="1.00403327015078E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/camera_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation/meshes/Arm_URDF_2025/camera_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="camera_joint"
    type="continuous">
    <origin
      xyz="0.0756084691443581 0.0366419123913177 0.133665837449048"
      rpy="0 0 0" />
    <parent
      link="link_5" />
    <child
      link="camera_link" />
    <axis
      xyz="0 1 0" />
  </joint>
</robot>
